### PR TITLE
update-reference-to-track2-for-azure-iot-central-arm-sdk-samples

### PR DESCRIPTION
**Pull request by colawwj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ The following table shows the various source repositories and package locations
 
 | Platform | Source                                        | Package                                                              |
 |----------|-----------------------------------------------|----------------------------------------------------------------------|
-| node     | https://github.com/Azure/azure-sdk-for-node   | https://www.npmjs.com/package/@azure/arm-iotcentral                  |
+| js     | https://github.com/Azure/azure-sdk-for-js   | https://www.npmjs.com/package/@azure/arm-iotcentral                  |
```
Comment (qiaozha) [inline README.md:54]: javascript

File: nodejs/package.json [hunk 2]
```diff
@@ -10,8 +10,8 @@
     "author": "",
     "license": "ISC",
     "dependencies": {
-        "@azure/arm-iotcentral": "^4.1.0",
-        "@azure/ms-rest-nodeauth": "^3.0.3",
+        "@azure/arm-iotcentral": "^6.0.0",
+        "@azure/identity": "^2.0.4",
         "azure-arm-resource": "^7.3.0",
```
Comment (qiaozha) [inline nodejs/package.json:15]: please change this one as well.

File: nodejs/package.json [hunk 3]
```diff
@@ -10,8 +10,8 @@
...
         "ms-rest-azure": "^3.0.0"
```
Comment (qiaozha) [inline nodejs/package.json:16]: remove this

**Review by qiaozha — APPROVED** [review]
LGTM

File: nodejs/index.ts [hunk 4]
```diff
@@ -48,25 +48,34 @@ async function retrieveAppInfo(client): Promise<IotCentralClient> {
...
 async function updateApp(client): Promise<IotCentralClient> {
-    const result = await client.apps.update(RESOURCEGROUPNAME, RESOURCENAME, UPDATEAPP);
+    const result = await client.apps.beginUpdateAndWait(RESOURCEGROUPNAME, RESOURCENAME, UPDATEAPP);
     console.log(result);
     return new Promise<IotCentralClient>(resolve => resolve(client));
 }
 
 async function listAllAppsByResourceGroup(client): Promise<IotCentralClient> {
-    const result = await client.apps.listByResourceGroup(RESOURCEGROUPNAME);
+    const result = [];
```
Comment (poanchen) [inline nodejs/index.ts:57]: does it not already return an array of resource groups?

**Review by poanchen — APPROVED** [review]

**Comment by poanchen** [comment]
LGTM

File: nodejs/index.ts (hunk 4 above)
Comment (qiaozha) [reply nodejs/index.ts:57]: in the new generation JS SDks, we will return a iterable iterator for list operations.

**Comment by qiaozha** [comment]
@poanchen if it looks good to you, can you help merge this PR ? we don't have the permission.

**Comment by poanchen** [comment]
Thanks for the change! @colawwj and thanks for the review @qiaozha

**Pull request merged**